### ISSUE TITLE
[pkgconf] Build error

### ISSUE THREAD

**Issue by koprok** [microsoft/vcpkg#27458]
**Host Environment**

- OS: [Windows]
- Compiler: MinGW GCC 8.1

**To Reproduce**

`./vcpkg install --triplet x64-mingw-dynamic pkgconf`

**Failure logs**

```
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:96 (message):
    Command failed: C:/GitLab-Runner/builds/q99Wq7Cg/0/common/vcpkg/downloads/tools/ninja/1.10.2-windows/ninja.exe install -v
    Working Directory: C:/GitLab-Runner/builds/q99Wq7Cg/0/common/vcpkg/buildtrees/pkgconf/x64-mingw-dynamic-dbg
    Error code: 1
    See logs for more information:
      C:\GitLab-Runner\builds\q99Wq7Cg\0\common\vcpkg\buildtrees\pkgconf\package-x64-mingw-dynamic-dbg-out.log
Call Stack (most recent call first):
  scripts/cmake/vcpkg_install_meson.cmake:33 (vcpkg_execute_required_process)
  ports/pkgconf/portfile.cmake:67 (vcpkg_install_meson)
  scripts/ports.cmake:147 (include)
```
[package-x64-mingw-dynamic-dbg-out.log](https://github.com/microsoft/vcpkg/files/9861260/package-x64-mingw-dynamic-dbg-out.log)
[meson-x64-mingw-dynamic-dbg.log](https://github.com/microsoft/vcpkg/files/9861263/meson-x64-mingw-dynamic-dbg.log)

**Additional context**

Hi, this issue appeared after upgrading from vcpkg tag 2022.09.27 to 2022.10.19.

I have investigated the issue a little bit. What I found is that `vcpkg_configure_meson()` removes the `-` prefix from all libraries passed to the linker, e.g. `-lkernel32` becomes `lkernel32`.
I hope this helps to fix this bug quickly.

**Comment by Neumann-A** [comment]
closed by #27380

**Comment by dg0yt** [comment]
Fix is merged now.
